src/utils/wallet.ts

```diff
@@ -505,36 +492,32 @@ export const fetchCoinsData = async ({
   refresh,
   quotes,
 }: {
-  coins: CoinBase[],
-  currentAccount: any,
-  vsCurrency: string,
-  currencyRate: number,
-  globalProps: GlobalProps,
-  quotes: { [key: string]: QuoteItem }
-  refresh: boolean,
-})
-  : Promise<{ [key: string]: CoinData }> => {
-
+  coins: CoinBase[];
+  currentAccount: any;
+  vsCurrency: string;
+  currencyRate: number;
+  globalProps: GlobalProps;
+  quotes: { [key: string]: QuoteItem };
+  refresh: boolean;
+}): Promise<{ [key: string]: CoinData }> => {
   const username = currentAccount.username;
-  const { base, quote, hivePerMVests } = globalProps
-
   const coinData = {} as { [key: string]: CoinData };
   const walletData = {} as any;
 
-
   if (!username) {
     return walletData;
   }
 
+  //fetch latest global props if refresh or data not available
+  const { base, quote, hivePerMVests } =
+    refresh || !globalProps || !globalProps.hivePerMVests ? await fetchGlobalProps() : globalProps;
```

Comment:
we can cache these globalProps also for couple minutes so it might lower number of requests. or useQuery for retries and cache...

Reply:
will be using useQuery for some the wallet related issue, but useQuery also have limitations, like it is only meant to be used somewhat directly with a component.